### PR TITLE
[17.0][MIG] sale_financial_risk

### PR DESCRIPTION
**Pull request by sbiosca-s73**
Migrate sale_financial_risk module to version 17.0

### CODE REVIEW

File: sale_financial_risk/models/res_partner.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ def _compute_risk_sale_order(self):
             domain=self._get_risk_sale_order_domain(),
             fields=["risk_partner_id", "company_id", "risk_amount"],
             groupby=["risk_partner_id", "company_id"],
-            orderby="id",
+            orderby="risk_partner_id, company_id",
```
Comment (carlosdauden) [inline sale_financial_risk/models/res_partner.py]: Why you need order by that fields?
For better performance use _read_group to avoid order.

**Comment by pedrobaeza** [comment]
https://github.com/OCA/maintainer-tools/wiki/Use-temporary-reference%28s%29-to-another-pull-request%28s%29

/ocabot migration sale_financial_risk

File: sale_financial_risk/models/sale.py [hunk 2]
```diff
@@ -49,7 +49,9 @@ def action_confirm(self):
                             {
                                 "exception_msg": exception_msg,
                                 "partner_id": partner.id,
-                                "origin_reference": "%s,%s" % ("sale.order", order.id),
+                                "origin_reference": "{},{}".format(
+                                    "sale.order", order.id
+                                ),
```
Comment (carlosdauden) [inline sale_financial_risk/models/sale.py]: You can replace by:
"origin_reference": f"{order._name},{order.id}",

**Review by carlosdauden — CHANGES_REQUESTED** [review]

**Comment by sbiosca-s73** [comment]
@carlosdauden changes done

File: sale_financial_risk/models/res_partner.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            partner = self.browse(risk_partner_id)
+            company = self.env["res.company"].browse(company_id or self.env.company.id)
+            company_currency = company.currency_id
+            partner.risk_sale_order = company_currency._convert(
+                risk_amount,
+                partner.risk_currency_id,
+                company,
+                fields.Date.context_today(self),
+                round=False,
+            )
```
Comment (carlosdauden) [inline sale_financial_risk/models/res_partner.py]: Try this:

```suggestion
        orders_group = self.env["sale.order.line"]._read_group(
            domain=self._get_risk_sale_order_domain(),
            groupby=["risk_partner_id", "company_id"],
            aggregates=["risk_amount:sum"],
        )
        for partner. company, risk_amount in orders_group:
            company_currency = company.currency_id
            partner.risk_sale_order = company_currency._convert(
                risk_amount,
                partner.risk_currency_id,
                company,
                fields.Date.context_today(self),
                round=False,
            )
```

**Review by carlosdauden — CHANGES_REQUESTED** [review]

**Comment by sbiosca-s73** [comment]
@carlosdauden changes done

**Comment by sbiosca-s73** [comment]
Not working test-requirements because the version of module account_financial_risk isn't correctly [version](https://github.com/OCA/credit-control/pull/347/files#diff-663023aec82d18b9e90463b1db577c8fef26638b6b60db1d62498dc4f55f34c5R7)

**Review by carlosdauden — APPROVED** [review]
Code LGTM

**Review by ioans73 — APPROVED** [review]

**Comment by Danisan** [comment]
@pedrobaeza good work!

**Comment by pedrobaeza** [comment]
You have to give thanks to the contributors doing the migration and the review.

@sbiosca-s73 please split the pre-commit changes from the migration itself as stated in the migration guide: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-17.0

**Comment by sbiosca-s73** [comment]
@pedrobaeza changes done. Thanks!

**Pull request merged**